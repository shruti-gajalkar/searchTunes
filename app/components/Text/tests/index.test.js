/**
 *
 * Tests for Text
 *
 */

import React from 'react';
import { renderWithIntl, getComponentStyles } from '@utils/testUtils';
import { Text } from '../index';

describe('<Text /> component tests', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<Text />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 Text component', () => {
    const { getAllByTestId } = renderWithIntl(<Text />);
    expect(getAllByTestId('text').length).toBe(1);
  });

  it('should contain render the text according to the id', () => {
    const { getAllByText } = renderWithIntl(<Text id="repo_list" />);
    expect(getAllByText(/Repository List/).length).toBe(1);
  });

  it('should have a margin-bottom of 5px when we pass marginBottom as 5', () => {
    const props = {
      marginBottom: 5,
      id: 'repo_list'
    };
    const styles = getComponentStyles(Text, props);
    expect(styles['margin-bottom']).toBe(`${props.marginBottom}px`);
  });
});
