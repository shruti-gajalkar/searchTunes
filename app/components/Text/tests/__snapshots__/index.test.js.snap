// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> component tests should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="Text__StyledText-ap7h6t-0 kCetqI"
      data-testid="text"
    >
      some_text
    </div>
  </div>
</body>
`;
